include/qore/intern/qore_thread_intern.h

```diff
@@ -418,40 +412,98 @@ class QoreParseClassHelper {
    DLLLOCAL ~QoreParseClassHelper();
 };
 
-class QoreProgramLocationHelper {
+class QoreProgramStackLocationHelper {
+public:
+    DLLLOCAL QoreProgramStackLocationHelper(QoreStackLocation* stack_loc, const AbstractStatement*& current_stmt,
+        QoreProgram*& current_pgm) :
```

Comment:
calling the ctor of `QoreProgramStackLocationHelper` changes the `current_stmt` and `current_pgm`? I don't get it why?

Reply:
it retrieves the current program and statement so they can be returned by the the `QoreStackLocation` using that object in its stack position - this functionality is used by the debugger.